### PR TITLE
Use cmake+ninja by default

### PR DESCRIPTION
**Pull request by th0ma7**
_Motivation:_  With recent cmake it is possible to use ninja to build which is more efficient than using default cmake.
_Linked issues:_  This also address weird build issue with #4797

### Checklist
- [x] Build rule `all-supported` completed successfully
- [ ] Package upgrade completed successfully
- [ ] New installation of package completed successfully

### CODE REVIEW

**Comment by th0ma7** [comment]
@hgy59 , @publicarray , @ymartin59 this is now good to go.
It solves build issues currently against master and uses ninja always with cmake.

File: cross/libaom/Makefile [hunk 1]
```diff
@@ -27,6 +27,9 @@ include ../../mk/spksrc.cross-cmake.mk
 # libaom specific cmake options
 CMAKE_ARGS += -DCONFIG_AV1_ENCODER=1
 
+# Disable tests
+CMAKE_ARGS += -DENABLE_TESTS=0
```
Comment (publicarray) [inline cross/libaom/Makefile:31]: Was it the tests that did not like being run in parallel? Otherwise, I wouldn't mind keeping them

**Comment by publicarray** [comment]
Nice, thanks for this! FYI Ninja is only 1 second or so better in compiling fish and umurmur.

File: cross/libaom/Makefile (hunk 1 above)
Comment (th0ma7) [reply cross/libaom/Makefile:31]: No, not related to parallel build (although at first I thought so).  With the usage of ninja all build failures went away.
I removed it as it reduces the compile time significantly (about 10%), went from 480+ down to 440ish ninja build actions.  Feel free to re-enable as you see fit.  Personally I have no need for it.

**Comment by th0ma7** [comment]
> Nice, thanks for this! FYI Ninja is only 1 second or so better in compiling fish and umurmur.

Thnx.  Indeed, it ain't faster, just more robust and efficient and it resolved all build failures I had from github action.  It also allowed me to fix cmake as it only had c & c++ calls as now it auto-detects all cross tools correctly automatically.

**Pull request merged**